src/machine/machinectl.c

```diff
@@ -163,20 +190,32 @@ static int get_os_name(sd_bus *bus, const char *name, char **out) {
         if (r < 0)
                 return bus_log_parse_error(r);
 
-        str = strdup(os);
-        if (!str)
-                return log_oom();
-        *out = str;
+        va_start(ap, query);
+        for (count = 0; count < awaited_args; count++) {
+                char *val, **out;
+
+                out = va_arg(ap, char **);
+                if (_unlikely_(!out))
+                        return -EINVAL;
```

Comment:
also, functions in systemd should either log about no errors (in which case the caller should log) or about all errors. This function so far opted for logging about all errors, but this one.

That said, if the parameter is NULL here, then that's really a programming error, not a runtime error. Hence `assert()` is the right choice here really, and not a friendly if check with return of EINVAL...